aggregator/src/datastore.rs

```diff
@@ -718,6 +704,94 @@ impl<C: Clock> Transaction<'_, C> {
         )?)
     }
 
+    /// Add an aggregator authentication token to the task with the given `task_id`.
+    pub async fn add_aggregator_authentication_token(
```

Comment:
OK. I think we should design the test API to require the least amount of effort / the least number of capabilities to be added to the "mainline" DAP implementation's functionality; the reason I think this is important is that requiring less effort to adopt the test API will make it more likely for the test API to be adopted.

In this case, there is no reason for a full implementation of DAP-02 to have the ability to modify task parameters -- DAP does not require one & the discussion amongst folks so far has leaned towards task parameters being considered immutable. (We will eventually need to introduce at least rotation for the parameters which are secret, but the details have not yet been decided.)

Given that, I would suggest a different strategy here. Can you speak more to why you prefer against "more optional attributes to `/internal/test/add_task`"? I would lean towards that strategy (concretely, perhaps I would make the test API task definition take a slice of enum "authenticator" values to support flexibly adding more authentication modes in the future; `serde_json` is smart enough to serialize/deserialize these values, and the test API would translate to the appropriate task parameters). Looking at the PR as it is, this implementation strategy would allow the new datastore methods to be dropped entirely; and the change to the test API test (`end_to_end.rs`) would require fewer changes.

(Also, apologies for not making this suggestion at the design stage. I did not have GH set up to watch the test API repo; I think I missed it since it lives in your personal account. I'll receive notifications for PRs against it from now on.)

Reply:
I opened https://github.com/divergentdave/draft-dcook-ppm-dap-interop-test-design/issues/28 to track this spec change